cql2pgjson/src/main/java/org/z3950/zing/cql/cql2pgjson/CQL2PgJSON.java

```diff
@@ -472,30 +472,56 @@ private SqlSelect toSql(CQLSortNode node) throws QueryValidationException {
       if (modifiers.cqlSort == CqlSort.DESCENDING) {
         desc = " DESC";
       }  // ASC not needed, it's Postgres' default
-
       IndexTextAndJsonValues vals = getIndexTextAndJsonValues(modifierSet.getBase());
-
+      //prepare sort by clause as if the sort by is not the primary key
+      String sortbyClauseText  = wrapInLowerUnaccent(vals.indexText);
+      String sortByClauseIndex = vals.indexJson;
+      //determine if sort is primary key 
+      
+      String pkColumnName = getPkColumnName();
+      String pkColumnComparator = pkColumnName;
+      String actualModifierBase = modifierSet.getBase();
+      //_id is actually referred to as id by cql so we need to check for the known cql value
+
+      if(actualModifierBase.equals("id")) {
+        actualModifierBase = pkColumnName;
+      }
+      if(pkColumnComparator != null && pkColumnComparator.equals(actualModifierBase)) {
+        //if it is we short circuit the json version of the column and just use the pkColumnName 
+        sortbyClauseText =  pkColumnName;
+        sortByClauseIndex = pkColumnName;
```

Comment:
If we know that we use pkColumnName there is no need to continue with testing NUMBER, "number" and "integer". Instead the pkColumnName should be appended to order and `continue` should be used to immediately skip to the next element of the for loop. Similar to the CqlSortType.NUMBER if-clause.